README.md

```diff
@@ -202,25 +207,91 @@ This will generate:
 - `dist/bindings/android/e4.aar`: the Android package, containing compiled Java class and native libraries for most common architectures
 - `dist/bindings/android/e4-sources.jar`: the Java source files
 
-After importing the AAR in your project, E4 client can be created and invoked 
+After importing the AAR in your project, E4 client can be created and invoked
 in a similar way than the Go version, for example using Kotlin:
 
 ```kotlin
-import io.teserakt.e4.*
-import io.teserakt.crypto.*
+import io.teserakt.e4.E4
+import io.teserakt.e4.SymNameAndPassword
+import io.teserakt.crypto.Crypto
 
 val cfg = SymNameAndPassword()
 cfg.name = "deviceXYZ"
 cfg.password = "secretForDeviceXYZ"
 
-val client = E4.newClient(cfg,  cacheDir.path + "deviceXYZ.json")
+val store = FileStore(File(filesDir.absolutePath + "/" + cfg.name + ".json"))
+val client = E4.newClient(cfg, store)
 
 // From here, messages can be protected / unprotected :
 val topic = "/deviceXYZ/data";
 val protectedMessage = client.protectMessage("Hello".toByteArray(Charsets.UTF_8), topic)
 val unprotectedMessage = client.unprotect(protectedMessage, topic)
 ```
 
+Here We are using a custom file storage implemented as such:
+```java
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+
+import io.teserakt.e4.ReadWriteSeeker;
+
+public class FileStore implements ReadWriteSeeker {
+    private static final int SEEK_START = 0;
+    private static final int SEEK_CURRENT = 1;
+    private static final int SEEK_END = 2;
+
+    private File file;
+    private int index;
+
+    public FileStore(File file) {
+        this.file = file;
+        this.index = 0;
+    }
+
+    public long read(byte[] buf) throws Exception {
+        FileInputStream fis = new FileInputStream(this.file);
+        int n = fis.read(buf, this.index, buf.length);
+        fis.close();
+
+        this.index += n;
+        return n;
+    }
+
+    public long write(byte[] buf) throws Exception {
+        FileOutputStream fos = new FileOutputStream(this.file);
```

Comment:
Ditto here about a single initialization in the constructor and no need to flush or close on every single write.